src/phpDocumentor/AutoloaderLocator.php

```diff
@@ -32,40 +32,31 @@ public static function autoload() : ClassLoader
      * This is the expected location if phpDocumentor is installed using composer because the current directory for
      * this file is expected to be 'vendor/phpdocumentor/phpdocumentor/src/phpDocumentor'.
      *
-     * If a composer.json is found we will try to extract the vendor folder name using the 'vendor-dir' configuration
-     * option of composer or assume it is vendor if that option is not set.
-     *
-     *
-     * If no custom composer.json can be found, then we assume that the vendor folder is that of phpDocumentor itself,
-     * which is `../../vendor` starting from this folder.
-     *
-     * If neither locations exist, then this method returns null because no vendor path could be found.
+     * If composer uses a custom composer.json filename with COMPOSER environment variable or a custom vendor directory
+     * name with COMPOSER_VENDOR_DIR or with a 'vendor-dir' entry in composer.json, the vendor folder won't be
+     * named 'vendor' so we base the matching on the presence of the autoload.php file generated by composer during
+     * install.
      *
+     * @see https://getcomposer.org/doc/03-cli.md#composer
+     * @see https://getcomposer.org/doc/03-cli.md#composer-vendor-dir
+     * 
      * @param string $baseDir parameter for test purposes only.
      */
     public static function findVendorPath(string $baseDir = __DIR__) : string
     {
         // default installation
         $vendorDir = $baseDir . '/../../vendor';
         // Composerised installation, vendor/phpdocumentor/phpdocumentor/src/phpDocumentor is __DIR__
-        $rootFolderWhenInstalledWithComposer = $baseDir . '/../../../../../';
-        $composerConfigurationPath = $rootFolderWhenInstalledWithComposer . 'composer.json';
-        if (file_exists($composerConfigurationPath)) {
-            $vendorDir = $rootFolderWhenInstalledWithComposer
-                . self::getCustomVendorPathFromComposer($composerConfigurationPath);
+        $vendorFolderWhenInstalledWithComposer = $baseDir . '/../../../../';
+        if (file_exists($vendorFolderWhenInstalledWithComposer.'/autoload.php')) {
+            $vendorDir = $vendorFolderWhenInstalledWithComposer;
         }
+        
+        $vendorDir = realpath($vendorDir);
```

Comment:
The use of realpath may interfere with running phpDocumentor's PHAR distribution. Realpath will break out of a PHAR file and consult the host filesystem. This is the reason why the original code did not use realpath to clean up the path :)

Reply:
I didn't know that, thanks. I just added realpath to ease the debug with more readability.

Thanks a lot

Reply:
No problem, I know this because I did the exact same thing when I originally built this class ;)

Reply:
I see you rebased the PR but the realpath is still there; do you still have to make the change or did something get lost in the rebase?

Reply:
I just wanted to see the failing tests with realpath rebased on master (because your approach for integration tests interests me) but Behat's tests are passing loacally (`80 scenarios (80 passed)
283 steps (283 passed)`).

I'm surprised they are not run automatically with this PR and I get a warning/error in the Github's actions of my commit: 
+ https://github.com/jclaveau/phpDocumentor/runs/366043339
+ https://github.com/jclaveau/phpDocumentor/runs/366043344

So I wonder if 
+ Behat's is wright locally but wrong in Github's actions?
+ The opposit.
+ The tests you speak about are still not merged in master?

Reply:
The failing build you are seeing is using the phar which has been built in a step before we execute behat. As @mvriel already said your changes are breaking the phar and not the plain build which is also passing in your linked build.

Does that make it more clear what is going on? 

We are still searching for a solution to build the pr on this project as well. So we can see the results in the pr. Main issue is that we don't want to build on pr and push but only one of them

Reply:
The failing build you are seeing is using the phar which has been built in a step before we execute behat. As @mvriel already said your changes are breaking the phar and not the plain build which is also passing in your linked build.

Does that make it more clear what is going on? 

We are still searching for a solution to build the pr on this project as well. So we can see the results in the pr. Main issue is that we don't want to build on pr and push but only one of them

Reply:
Thanks for the explanations. I removed the realpath() call